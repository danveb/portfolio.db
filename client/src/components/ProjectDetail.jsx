import { useState, useEffect } from "react"; 
import { useLocation } from "react-router-dom";
import Button from "./Button";
import { projects } from "../constants/projects";
import "../styles/ProjectDetail.css"; 

export default function ProjectDetail() {
  // useLocation
  // grab location from current param for name of "project" 
  const location = useLocation(); 
  const path = location.pathname.split("/")[2]; 

  // useState 
  const [project, setProject] = useState([]); 

  // useEffect
  useEffect(() => {
    try {
      // filter the project solely based on title of the project, which eventually matches to current path
      let filteredProject = projects.filter((project) => project.title === path); 
      // save filtered project to project "state"
      setProject(filteredProject); 
    } catch (error) {
      console.log(error); 
    }
  }, [path]); 

  return (
    <div className="projectDetail" data-testid="projectDetail">
      <div className="projectDetail__wrapper">
        <div className="projectDetail__main">
          {project.map((p) => (
            <div key={p.id}>
              <section className="projectDetail__top">
                <h1>{p.title}<span className="dot--green">.</span></h1>
                <div className="breakline"></div>
                <p>{p.description}</p>
                <Button title={p.liveUrlTitle} url={p.liveUrl} target="_blank" rel="noreferrer noopener"/>
                <Button title={p.gitHubTitle} url={p.gitHubUrl} target="_blank" rel="noreferrer noopener"/>
              </section>
              <section className="projectDetail__bottom">
                <img src={p.img} alt={p.alt} />
              </section>
            </div>
          ))}
        </div>
      </div>
    </div>
  )
}