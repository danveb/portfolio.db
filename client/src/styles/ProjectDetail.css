.projectDetail {
  width: 100%; 
  height: 100%; 
}

.projectDetail__wrapper {
  max-width: 768px; 
  margin: 0 auto; 
}

.projectDetail__main {
  height: 100%; 
  width: 100%; 
  display: flex; 
  flex-direction: column; 
  justify-content: center; 
  align-items: center; 
}

.projectDetail__top {
  display: flex; 
  flex-direction: column; 
  justify-content: center; 
  align-items: space-between; 
  gap: 30px; 
  margin: 40px 0; 
}

.projectDetail__top h1 {
  font-family: inherit; 
  font-size: 26px; 
  letter-spacing: 0.5px; 
}

.projectDetail__top p {
  font-family: inherit;
  font-size: 16px; 
  font-weight: 300; 
  color: #d4d4d4;
  letter-spacing: 1px; 
  line-height: 1.5rem; 
}

.projectDetail__bottom {
  margin-bottom: 40px; 
}

.projectDetail__bottom img {
  width: 100%; 
  height: 100%; 
  object-fit: contain; 
}

@media screen and (max-width: 1024px) {
  .projectDetail__wrapper {
    padding: 0 20px; 
  }
}

@media screen and (max-width: 768px) {
  .projectDetail__top h1 {
    font-size: 16px; 
  }

  .projectDetail__top p {
    font-size: 14px; 
  }
}