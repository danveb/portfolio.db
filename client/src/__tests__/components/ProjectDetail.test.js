import { render, screen } from "@testing-library/react"; 
import ProjectDetail from "../../components/ProjectDetail"; 
import { BrowserRouter } from "react-router-dom";

describe("ProjectDetail component", () => {
  // smoke test
  test("renders without crashing", () => {
    render(
      <BrowserRouter>
        <ProjectDetail />
      </BrowserRouter>
    ); 
  }); 

  // snapshot test
  test("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <ProjectDetail />
      </BrowserRouter>
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  });

  // getByTestId
  test("matches test id", () => {
    render(
      <BrowserRouter>
        <ProjectDetail />
      </BrowserRouter>
    );
    const testId = screen.getByTestId("projectDetail"); 
    expect(testId).toBeInTheDocument(); 
    expect(testId).not.toBe("Project Detail"); 
  });
});