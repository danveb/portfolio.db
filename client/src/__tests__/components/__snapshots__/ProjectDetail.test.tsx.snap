// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDetail component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="projectDetail"
    data-testid="projectDetail"
  >
    <div
      class="projectDetail__wrapper"
    >
      <div
        class="projectDetail__main"
      />
    </div>
  </div>
</DocumentFragment>
`;
